src/traces/histogram/calc.js

```diff
@@ -124,10 +132,17 @@ function calc(gd, trace) {
             rightGap = Math.min(rightGap, binEdges[n + 1] - posi);
         }
     }
+    roundFnOpts.leftGap = leftGap;
+    roundFnOpts.rightGap = rightGap;
 
     var roundFn;
     if(!uniqueValsPerBin) {
-        roundFn = getBinSpanLabelRound(leftGap, rightGap, binEdges, pa, calendar);
+        roundFn = function(v, isRightEdge) {
+            return function() {
+                var roundFnOpts = gd._fullLayout._roundFnOpts[groupName];
+                return getBinSpanLabelRound(roundFnOpts.leftGap, roundFnOpts.rightGap, binEdges, pa, calendar)(v, isRightEdge);
```

Comment:
This line is getting little too long. Maybe:
```js
return getBinSpanLabelRound(
  roundFnOpts.leftGap, 
  roundFnOpts.rightGap, 
  binEdges, pa, calendar
)(v, isRightEdge);
```

Reply:
Done in c387f9b74050731396044ec1bd381073fdf0799e